src/analyzer/plugins/buffering_utils.cpp

```diff
@@ -1,15 +1,25 @@
 #include "analyzer/plugins/buffering_utils.h"
 
 #include "util/math.h"
+#include "util/sample.h"
+
+#include <string.h>
 
 namespace mixxx {
 
 bool DownmixAndOverlapHelper::initialize(size_t windowSize, size_t stepSize, WindowReadyCallback callback) {
     m_buffer.resize(windowSize);
+    memset(m_buffer.data(), 0, sizeof(m_buffer[0]) * windowSize);
     m_callback = callback;
     m_windowSize = windowSize;
     m_stepSize = stepSize;
-    m_bufferWritePosition = 0;
+    // make sure the first frame is centered into the fft window. This makes sure
+    // that the result is significant starting fom the first step.
+    if (stepSize < windowSize) {
```

Comment:
I don't think that this calculation is correct. The mapping is discontinuous at stepSize == windowSize.

Without a deeper studying of the code I suppose that either `m_bufferWritePosition = windowSize - stepSize` or `m_bufferWritePosition = windowSize - stepSize / 2` should be used. Both result in 0 for the special case stepSize == windowSize.

Reply:
Mmm, you are right. But I am unsure how the solution will work.

With the current qm key analyzer we have actually the case "stepSize == windowSize".
With the given hop size of 8, it only uses 1 of 8 samples. If I include this case, it will pick an other set of "random" samples, altering the analysis result again. Would this be OK?

I think anyway, we may want to invalidate all key analysis before the release, because after all these rounding fixes the original qm algorithm is significantly improved.
My algorithm is already promising, and I don't want to invalidate the key analysis again mmm

What do you think?

Reply:
Random sampling would be inappropriate if it causes major result variations. This should only happen for rare edge cases.

Just before releasing the new version of the algorithm we should think about modifying the plugin id to trigger a re-analysis if enabled.